src/core.ts

```diff
@@ -0,0 +1,364 @@
+
+// main
+import { valuesOf } from './type-utils';
+
+// local
+import { ConfigurablePurpose, SpecialTrackingPurpose } from './enums';
+
+/* eslint-disable max-lines */
+
+/** Transcend logger items */
+export type LogItem = {
+  /** Log item content */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  content: any;
+  /** Log item styles */
+  styles: string[];
+};
+
+/** Transcend logger entry tag */
+export type LogTag = LogItem & {
+  /** Log tag content must be a string */
+  content: string;
+  /** Log tags must have only one style */
+  styles: string[1];
+};
+
+/** Transcend logger entry message */
+export type LogMessage = LogItem;
+
+/** Console-safe log levels */
+export const ConsoleSafeLogLevel = t.keyof({
+  info: null,
+  log: null,
+  warn: null,
+  debug: null,
+  error: null,
+  trace: null,
+  // meta-levels:
+  group: null,
+  groupCollapsed: null,
+  groupEnd: null,
+});
+
+/** Override type */
+export type ConsoleSafeLogLevel = t.TypeOf<typeof ConsoleSafeLogLevel>;
+
+/** Airgap log levels */
+export const LogLevel = t.union([ConsoleSafeLogLevel, t.literal('fatal')]);
+/** Override type */
+export type LogLevel = t.TypeOf<typeof LogLevel>;
+
+/** Airgap logger entry types */
+export type LogEntryType = LogLevel;
+
+/** Airgap logger entries */
+export type LogEntry = {
+  /** Log entry tag */
+  tag: LogTag;
+  /** Log entry message */
+  message: LogMessage;
+};
+
+/**
+ * Log emitter function
+ */
+export type LogEmitter = ((...entries: any[]) => void) & {
+  /** Styled log emitter function */
+  styled(styles?: null | string | string[], ...entries: any[]): void;
+};
+
+/** Transcend Logger API */
+export type Logger = {
+  /**
+   * Set log tag during callback execution or from
+   * this point on if no callback is provided.
+   */
+  tag(logTag: string, callback?: () => any): void;
+} & {
+  /** Log emitter (e.g. `logger.log()`) */
+  [method in LogLevel]: LogEmitter;
+};
+
+/** AirgapAuth auth options */
+export type AirgapAuthMap = {
+  /** A `load` event from a just-loaded airgap.js script element (which implies auth by being loaded before airgap.js) */
+  load?: Event;
+  /** A user-initiated interaction event that was just dispatched. */
+  interaction?: UIEvent;
+  /** Automatically reload the page if needed to remove CSP. `false` by default. */
+  autoReload?: boolean;
+};
+
+/** Airgap authorization proof */
+export type AirgapAuth =
+  | null
+  | AirgapAuthMap
+  /** A user-initiated interaction event that was just dispatched. */
+  | UIEvent
+  /** A `load` event from a just-loaded airgap.js script element (which implies auth by being loaded before airgap.js) */
+  | Event;
+
+/** A boolean value represented as either `'on'` or `'off'` */
+export const BooleanString = t.keyof({
+  on: null,
+  off: null,
+});
+/** Type override */
+export type BooleanString = t.TypeOf<typeof BooleanString>;
+
+/** Potentially applicable data privacy legal regimes */
+export const PrivacyRegime = t.union([
```

Comment:
do we need this? cant we just use `valuesOf(PrivacyRegimeEnum)`?

Reply:
yeah, that works too. I'll fix that in a follow-up pr as I've already merged this one